Quorum node crashing when running with QBFT

#### System information

Geth version: Geth
Version: 1.10.2-stable
Git Commit: 0b0881aa80b1fc03868c00adf936eb53d8402d95
Git Commit Date: 20220822
Quorum Version: 22.7.0
Architecture: amd64
Network Id: 1337
Go Version: go1.18.2
Operating System: darwin
GOPATH=
GOROOT=go
OS & Version: OSX
Commit hash : 0b0881aa80b1fc03868c00adf936eb53d8402d95

#### Expected behaviour
QBFT consensus working fine without any faulires

#### Actual behaviour
Node crashed with the below message:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x49d2292]

#### Steps to reproduce the behaviour
- bring up 7 node quorum network
- execute a smart contract running a simple for loop with high loop count
- tx fails with vm out of gas error 
- node crashes suddenly

#### Backtrace

````
INFO [09-01|16:38:00.912] QBFT: start new round                    address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 old.round=0  old.sequence=128 old.state=Committed        old.proposer=0xF512a992F3fb749857d758fFDa1330e590fa915E next.round=0 next.seq=129 next.proposer=0xb131288F355BC27090E542aE0be213c20350B767 next.valSet="[0x6571D97f340c8495B661a823F2C2145cA47D63c2 0x8157D4437104e3B8dF4451a85F7B2438ef6699FF 0xF512a992F3fb749857d758fFDa1330e590fa915E 0xb131288F355BC27090E542aE0be213c20350B767 0xb912De287F9b047B4228436E94B5b78E3Ee16171 0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 0xe36cbeB565B061217930767886474e3cDe903AC5]" next.size=7 next.IsProposer=false
INFO [09-01|16:38:26.756] VM returned with error                   err="out of gas"
INFO [09-01|16:38:26.756] Aborting transaction processing due to 'commitInterruptNewHead', elapsed time=30.864690149s
INFO [09-01|16:38:26.757] Commit new mining work                   number=129 sealhash="38586b…451231" uncles=0 txs=0 gas=0 fees=0 elapsed="229.351µs"
INFO [09-01|16:38:26.757] QBFT: handle block proposal request      address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state="Accept request"
INFO [09-01|16:38:30.832] QBFT: handle PRE-PREPARE message         address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966
INFO [09-01|16:38:30.833] QBFT: accepted PRE-PREPARE message       address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966
INFO [09-01|16:38:30.833] QBFT: changed state                      address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 old.state="Accept request" new.state=Preprepared
INFO [09-01|16:38:30.833] QBFT: broadcast PREPARE message          address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Preprepared      msg.code=19 msg.source=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 msg.round=0 msg.sequence=129 payload=0xf868e4818180a0a5c2d86eab596cece47b350312a86bdbd5dffa6979d2b3c838ef055ebbc390d2b841ef0297e53ad30632b6e5d56cbf33caeae079237b83ee3e80eb37233d9ea5f1e459e33e07d5f6581c92da2018ee313272ed2743c46dd566def7374ca344ebc95401
INFO [09-01|16:38:30.833] QBFT: handle PREPARE message             address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Preprepared      msg.code=19 msg.source=0xb131288F355BC27090E542aE0be213c20350B767 msg.round=0 msg.sequence=129 prepares.count=0 quorum=5
INFO [09-01|16:38:30.834] QBFT: handle PREPARE message             address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Preprepared      msg.code=19 msg.source=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 msg.round=0 msg.sequence=129 prepares.count=1 quorum=5
INFO [09-01|16:38:30.835] QBFT: handle PREPARE message             address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Preprepared      msg.code=19 msg.source=0xe36cbeB565B061217930767886474e3cDe903AC5 msg.round=0 msg.sequence=129 prepares.count=2 quorum=5
INFO [09-01|16:38:30.836] QBFT: handle PREPARE message             address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Preprepared      msg.code=19 msg.source=0x6571D97f340c8495B661a823F2C2145cA47D63c2 msg.round=0 msg.sequence=129 prepares.count=3 quorum=5
INFO [09-01|16:38:30.845] QBFT: handle PREPARE message             address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Preprepared      msg.code=19 msg.source=0xF512a992F3fb749857d758fFDa1330e590fa915E msg.round=0 msg.sequence=129 prepares.count=4 quorum=5
INFO [09-01|16:38:30.846] QBFT: received quorum of PREPARE messages address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Preprepared      msg.code=19 msg.source=0xF512a992F3fb749857d758fFDa1330e590fa915E msg.round=0 msg.sequence=129 prepares.count=5 quorum=5
INFO [09-01|16:38:30.846] QBFT: changed state                      address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 old.state=Preprepared      new.state=Prepared
INFO [09-01|16:38:30.846] QBFT: broadcast COMMIT message           address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Prepared         msg.code=20 msg.source=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 msg.round=0 msg.sequence=129 payload=0xf8acf867818180a0a5c2d86eab596cece47b350312a86bdbd5dffa6979d2b3c838ef055ebbc390d2b84124777982e8e519e9034b14694cb52a58c2298a398dcb45109a0adc2306717d304e0612d29bca4f8a7892a7b49704d5c989e4446b65cf0de016ac5546e30579e401b841dc8a7c9e7dbc3756e18a83239ed90e30aecc28d8005ab3c90faf4a3cd709144c4840406b4138cf4afca944f92de0824deafa029d11df2112541e87c6b392c2c601
INFO [09-01|16:38:30.849] QBFT: handle COMMIT message              address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Prepared         msg.code=20 msg.source=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 msg.round=0 msg.sequence=129 commits.count=0 quorum=5
INFO [09-01|16:38:30.850] QBFT: handle COMMIT message              address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Prepared         msg.code=20 msg.source=0xb912De287F9b047B4228436E94B5b78E3Ee16171 msg.round=0 msg.sequence=129 commits.count=1 quorum=5
INFO [09-01|16:38:30.850] QBFT: handle COMMIT message              address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Prepared         msg.code=20 msg.source=0xe36cbeB565B061217930767886474e3cDe903AC5 msg.round=0 msg.sequence=129 commits.count=2 quorum=5
INFO [09-01|16:38:30.851] QBFT: handle COMMIT message              address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Prepared         msg.code=20 msg.source=0x6571D97f340c8495B661a823F2C2145cA47D63c2 msg.round=0 msg.sequence=129 commits.count=3 quorum=5
INFO [09-01|16:38:30.852] QBFT: handle COMMIT message              address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Prepared         msg.code=20 msg.source=0xF512a992F3fb749857d758fFDa1330e590fa915E msg.round=0 msg.sequence=129 commits.count=4 quorum=5
INFO [09-01|16:38:30.852] QBFT: received quorum of COMMIT messages address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Prepared         msg.code=20 msg.source=0xF512a992F3fb749857d758fFDa1330e590fa915E msg.round=0 msg.sequence=129 commits.count=5 quorum=5
INFO [09-01|16:38:30.852] QBFT: changed state                      address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 old.state=Prepared         new.state=Committed
INFO [09-01|16:38:30.852] BFT: block proposal committed            author=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 hash="a5c2d8…c390d2" number=129
INFO [09-01|16:38:30.855] Imported new chain segment               blocks=1 txs=0 mgas=0.000 elapsed=1.330ms     mgasps=0.000 number=129 hash="a5c2d8…c390d2" dirty=0.00B
INFO [09-01|16:38:30.856] QBFT: handle final committed             address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 state=Committed
INFO [09-01|16:38:30.856] QBFT: initialize new round               address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=129 target.round=0 lastProposal.number=129 lastProposal.hash="a5c2d8…c390d2"
INFO [09-01|16:38:30.856] QBFT: changed state                      address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=130 old.state=Committed        new.state="Accept request"
INFO [09-01|16:38:30.856] QBFT: start new round                    address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 old.round=0  old.sequence=129 old.state=Committed        old.proposer=0xb131288F355BC27090E542aE0be213c20350B767 next.round=0 next.seq=130 next.proposer=0xb912De287F9b047B4228436E94B5b78E3Ee16171 next.valSet="[0x6571D97f340c8495B661a823F2C2145cA47D63c2 0x8157D4437104e3B8dF4451a85F7B2438ef6699FF 0xF512a992F3fb749857d758fFDa1330e590fa915E 0xb131288F355BC27090E542aE0be213c20350B767 0xb912De287F9b047B4228436E94B5b78E3Ee16171 0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 0xe36cbeB565B061217930767886474e3cDe903AC5]" next.size=7 next.IsProposer=false
INFO [09-01|16:38:41.235] QBFT: handle ROUND-CHANGE message        address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=130 state="Accept request" msg.code=21 msg.source=0xe36cbeB565B061217930767886474e3cDe903AC5 msg.round=1 msg.sequence=130 higherRoundChanges.count=0 currentRoundChanges.count=0
INFO [09-01|16:38:41.256] QBFT: handle ROUND-CHANGE message        address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=130 state="Accept request" msg.code=21 msg.source=0x6571D97f340c8495B661a823F2C2145cA47D63c2 msg.round=1 msg.sequence=130 higherRoundChanges.count=1 currentRoundChanges.count=0
INFO [09-01|16:38:41.285] QBFT: handle ROUND-CHANGE message        address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=130 state="Accept request" msg.code=21 msg.source=0x8157D4437104e3B8dF4451a85F7B2438ef6699FF msg.round=1 msg.sequence=130 higherRoundChanges.count=2 currentRoundChanges.count=0
INFO [09-01|16:38:41.286] QBFT: received F+1 ROUND-CHANGE messages address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=130 state="Accept request" msg.code=21 msg.source=0x8157D4437104e3B8dF4451a85F7B2438ef6699FF msg.round=1 msg.sequence=130 higherRoundChanges.count=3 currentRoundChanges.count=0 F=2
INFO [09-01|16:38:41.286] QBFT: initialize new round               address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=0 current.sequence=130 target.round=1 lastProposal.number=129 lastProposal.hash="a5c2d8…c390d2"
INFO [09-01|16:38:41.286] QBFT: start new round                    address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 old.round=0  old.sequence=130 old.state="Accept request" old.proposer=0xb912De287F9b047B4228436E94B5b78E3Ee16171 next.round=1 next.seq=130 next.proposer=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 next.valSet="[0x6571D97f340c8495B661a823F2C2145cA47D63c2 0x8157D4437104e3B8dF4451a85F7B2438ef6699FF 0xF512a992F3fb749857d758fFDa1330e590fa915E 0xb131288F355BC27090E542aE0be213c20350B767 0xb912De287F9b047B4228436E94B5b78E3Ee16171 0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 0xe36cbeB565B061217930767886474e3cDe903AC5]" next.size=7 next.IsProposer=true
INFO [09-01|16:38:41.286] QBFT: broadcast ROUND-CHANGE message     address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=1 current.sequence=130 state="Accept request" msg.code=21 msg.source=0x0000000000000000000000000000000000000000 msg.round=1 msg.sequence=130 payload=0xf84cf848c4818201c0b8413b079a705d70c2cf4c6487b4081b89ceb31b11425d03f80aa2968290437d2c48478fad9dd4c40980047c5682dc96889e6e27eb9140eccad1e31af02fca6adc9301c0c0
INFO [09-01|16:38:41.287] QBFT: handle ROUND-CHANGE message        address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=1 current.sequence=130 state="Accept request" msg.code=21 msg.source=0xF512a992F3fb749857d758fFDa1330e590fa915E msg.round=1 msg.sequence=130 higherRoundChanges.count=0 currentRoundChanges.count=3
INFO [09-01|16:38:41.287] QBFT: handle ROUND-CHANGE message        address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=1 current.sequence=130 state="Accept request" msg.code=21 msg.source=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 msg.round=1 msg.sequence=130 higherRoundChanges.count=0 currentRoundChanges.count=4
INFO [09-01|16:38:41.287] QBFT: received quorum of ROUND-CHANGE messages address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 current.round=1 current.sequence=130 state="Accept request" msg.code=21 msg.source=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 msg.round=1 msg.sequence=130 higherRoundChanges.count=0 currentRoundChanges.count=5
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x49d2292]

goroutine 338 [running]:
github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core.(*core).handleRoundChange(0xc0002a4100, 0xc00172cb40)
	github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core/roundchange.go:153 +0x652
github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core.(*core).deliverMessage(0xc0002a4100, {0x555f0e0, 0xc00172cb40})
	github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core/handler.go:217 +0xf8
github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core.(*core).handleDecodedMessage(0xc0002a4100?, {0x555f0e0, 0xc00172cb40})
	github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core/handler.go:202 +0xd1
github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core.(*core).handleEncodedMsg(0xc0002a4100, 0xc0009e3f2c?, {0xc00164c0f0, 0x4e, 0x4e})
	github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core/handler.go:188 +0x385
github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core.(*core).handleEvents(0xc0002a4100)
	github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core/handler.go:122 +0x2dd
created by github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core.(*core).Start
	github.com/ethereum/go-ethereum/consensus/istanbul/qbft/core/handler.go:37 +0xa5
````

When submitting logs: please submit them as text and not screenshots.
[1.log](https://github.com/ConsenSys/quorum/files/9474590/1.log)

--- a/consensus/istanbul/qbft/core/core.go
+++ b/consensus/istanbul/qbft/core/core.go
@@ -189,7 +189,7 @@ func (c *core) startNewRound(round *big.Int) {
 	c.valSet.CalcProposer(lastProposer, newView.Round.Uint64())
 	c.setState(StateAcceptRequest)
 
-	if round.Cmp(c.current.Round()) > 0 {
+	if c.current != nil && round.Cmp(c.current.Round()) > 0 {
 		roundMeter.Mark(new(big.Int).Sub(round, c.current.Round()).Int64())
 	}
 

--- a/consensus/istanbul/qbft/core/handler.go
+++ b/consensus/istanbul/qbft/core/handler.go
@@ -286,9 +286,12 @@ func (c *core) verifySignatures(m qbfttypes.QBFTMessage) error {
 }
 
 func (c *core) currentLogger(state bool, msg qbfttypes.QBFTMessage) log.Logger {
-	logCtx := []interface{}{
-		"current.round", c.current.Round().Uint64(),
-		"current.sequence", c.current.Sequence().Uint64(),
+	logCtx := []interface{}{}
+	if c.current != nil {
+		logCtx = append(logCtx,
+			"current.round", c.current.Round().Uint64(),
+			"current.sequence", c.current.Sequence().Uint64(),
+		)
 	}
 
 	if state {

--- a/consensus/istanbul/qbft/core/roundchange.go
+++ b/consensus/istanbul/qbft/core/roundchange.go
@@ -17,6 +17,7 @@
 package core
 
 import (
+	"errors"
 	"math/big"
 	"sort"
 	"sync"
@@ -26,6 +27,7 @@ import (
 	"github.com/ethereum/go-ethereum/consensus/istanbul"
 	qbfttypes "github.com/ethereum/go-ethereum/consensus/istanbul/qbft/types"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
@@ -150,7 +152,12 @@ func (c *core) handleRoundChange(roundChange *qbfttypes.RoundChange) error {
 		// propose the block proposal that we generated
 		_, proposal := c.highestPrepared(currentRound)
 		if proposal == nil {
-			proposal = c.current.pendingRequest.Proposal
+			if c.current != nil && c.current.pendingRequest != nil {
+				proposal = c.current.pendingRequest.Proposal
+			} else {
+				log.Warn("round change returns an error: no proposal as pending request is nil")
+				return errors.New("no proposal as pending request is nil")
+			}
 		}
 
 		// Prepare justification for ROUND-CHANGE messages

